src/utils/CaminoExporter.ts

```diff
@@ -0,0 +1,81 @@
+import { Pathway } from 'pathways-model';
+import { Criteria } from 'criteria-model';
+import { v4 as uuidv4 } from 'uuid';
+import { deepCopyPathway } from './nodeUtils';
+import { constructCqlLibrary, IncludedCqlLibraries } from './export';
+
+export class CaminoExporter {
+  pathway: Pathway;
+  criteria: Criteria[];
+
+  constructor(pathway: Pathway, criteria: Criteria[]) {
+    this.pathway = deepCopyPathway(pathway);
+    this.criteria = criteria;
+  }
+
+  export(): Pathway {
+    // goal here is to iterate through the pathway,
+    // find all referenced libraries,
+    // and update the CQL in the transition to point to them appropriately
+    const libraryId = uuidv4();
+    const libraryName = `LIB${libraryId.substring(0, 7)}`;
+
+    const includedCqlLibraries: IncludedCqlLibraries = {};
+    const referencedDefines: Record<string, string> = {};
+
+    const builderDefines: Record<string, string> = {};
+
+    // iterate through the nodes and find all criteria that are actually used.
+    // for each one, add the criteria CQL to our appropriate map
+    //  - if it's an included library, keep track of the library name and version
+    //    as well as the specific definition we're referencing
+    //  - if it was constructed in the builder, track the name and raw CQL
+    for (const nodeId in this.pathway.nodes) {
+      const node = this.pathway.nodes[nodeId];
+      for (const transition of node.transitions) {
+        if (transition.condition?.criteriaSource) {
+          const criteriaSource = this.criteria.find(
+            c => c.id === transition.condition?.criteriaSource
+          );
+          if (criteriaSource?.elm && criteriaSource?.cql) {
+            const libraryIdentifier = criteriaSource.elm.library.identifier;
+            includedCqlLibraries[libraryIdentifier.id] = {
+              cql: criteriaSource.cql,
+              version: libraryIdentifier.version
+            };
+
+            referencedDefines[transition.condition.cql] = libraryIdentifier.id;
+
+            // prepend the library name if not already done
+            if (!transition.condition.cql.startsWith(libraryIdentifier.id)) {
```

Comment:
Technically I think there's a chance of false positives here, but more importantly I would have expected the importer to remove the library id for consistency. Nothing else breaks when it's present after importing?

Reply:
I didn't see anything else break because most of the other functionality is keyed off `criteriaSource` as far as I know.  I'll make the change you suggested.

Reply:
After talking with Reece, it seems strange that nothing breaks, and apparently it's a lot easier to do this than to remove it as part of the import process, so in the interest of getting this in soon, we'll leave this as is

Reply:
So I started working on changing this, but its a little trickier than I expected.

Right now re-associating the CQL requires that the library be prefixed onto the CQL so that it can find the criteria that matches the library/statement combo.  All the existing sample pathways are organized this way too.  I think we'd need to modify the data model to store the information linking the CQL to the library otherwise.  This information has to left on the pathway and I can't have the importer remove it because the re-association is done in a `useEffect` hook after the fact because setting the new criteria is asynchronous.